ankihub/gui/config_dialog.py

```diff
@@ -66,3 +75,42 @@ def _general_tab(conf_window) -> None:
     tab.checkbox("debug_level_logs", "Verbose logs (restart required)")
 
     tab.stretch()
+
+
+def add_nested_checkboxes(config_layout, key_prefix: str, description: str) -> None:
+
+    from .ankiaddonconfig.window import ConfigLayout
+
+    config_layout = cast(ConfigLayout, config_layout)
+
+    main_checkbox = QCheckBox(description)
+    config_layout.addWidget(main_checkbox)
+
+    container_outer = config_layout.hcontainer()
+    container_outer.setContentsMargins(0, 2, 0, 2)
+
+    container_inner = container_outer.vcontainer()
+    container_inner.setContentsMargins(30, 0, 0, 0)
+
+    step_1_checkbox = container_inner.checkbox(
+        f"{key_prefix}_step_1", description="USMLE Step 1"
+    )
+    step_2_checkbox = container_inner.checkbox(
+        f"{key_prefix}_step_2", description="USMLE Step 2"
+    )
+
+    def update_main_checkbox() -> None:
+        main_checkbox.blockSignals(True)
+        main_checkbox.setChecked(
```

Comment:
Maybe we can set Qt.CheckState.PartiallyChecked if only one step is enabled?

Reply:
Good idea!